src/Microsoft.DotNet.VersionTools/Automation/PullRequestCreator.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.VersionTools.Automation.GitHubApi;
+using Microsoft.DotNet.VersionTools.Util;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.VersionTools.Automation
+{
+    public class PullRequestCreator
+    {
+        private GitHubAuth _auth;
+
+        public string BranchPrefix { get; set; } = "UpdateDependencies";
+
+        public GitHubBranch UpstreamBranch { get; }
+        public GitHubProject Origin { get; }
+
+        public string GitAuthorName { get; }
+
+        public PullRequestCreator(
+            GitHubAuth auth,
+            string projectName,
+            string upstreamOwner,
+            string upstreamBranch,
+            string gitAuthorName)
+            : this(
+                auth,
+                new GitHubProject(projectName, auth.User),
+                new GitHubBranch(upstreamBranch, new GitHubProject(projectName, upstreamOwner)),
+                gitAuthorName)
+        {
+        }
+
+        public PullRequestCreator(
+            GitHubAuth auth,
+            GitHubProject origin,
+            GitHubBranch upstreamBranch,
+            string gitAuthorName = null)
+        {
+            _auth = auth;
+
+            Origin = origin;
+            UpstreamBranch = upstreamBranch;
+
+            GitAuthorName = gitAuthorName ?? auth.User;
+        }
+
+        public async Task CreateOrUpdateAsync(
+            string commitMessage,
+            string title,
+            string description,
+            bool forceCreate = false)
+        {
+            using (var client = new GitHubClient(_auth))
+            {
+                var originBranch = new GitHubBranch(
+                    $"{BranchPrefix}{DateTime.UtcNow.ToString("yyyyMMddhhmmss")}",
```

Comment:
Hmm, realized that we do need a different upgrade branch for each target branch in the repo. `master`, `rel/1.0.0`, `dev/api`, etc. need separate branches per-PR. Doable, but we'd need to bring in that info somehow--another parameter passed in by the Maestro build, taken from VSTS environment, or another `git` command are the ways I can think of.

@eerhardt Didn't see your comment when I posted mine earlier (I must not have refreshed), but yeah, that could do it. If we want to go with a `git` command or the environment we could pass the branch name into the strategy.